include/LLGL/Platform/Linux/LinuxNativeHandle.h

```diff
@@ -36,6 +37,37 @@ struct NativeHandle
     int             screen;
 };
 
+#ifdef LLGL_LINUX_ENABLE_WAYLAND
+
+struct WaylandNativeHandle
+{
+    struct wl_surface* window;
+    struct wl_display* display;
+};
+
+#else
+
+struct WaylandNativeHandle{};
+
+#endif
+
+enum class NativeHandleType : char
+{
+    X11 = 0,
+    Wayland = 1
+};
+
+//! Linux native handle structure.
+struct NativeHandle
+{
+    union {
+        X11NativeHandle x11;
```

Comment:
We have the same type of deprecation problem here. How about this, keep this union as is, but also keep the old members inside this `NativeHandle` struct. When an X11 window is used, we initialize both the old members and the `x11` field and when Wayland is used we initialize only the `wayland` field and reset the rest to zero. So something like this:
```cpp
struct NativeHandle
{
  LLGL_DEPRECATED("Deprecated since 0.04b; Use x11.display instead.")
  ::Display* display;

  ...

  union
  {
    X11NativeHandle x11;
    WaylandNativeHandle wayland;
  }
};
```